### PR TITLE
feat: Enforce password meets zxcvbn standard

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

If a user wants to enter a bad password - their problem right? WRONG. We already indicate password strength. Lets enforce it to help people help themselves

## Changes

* Adds validation to all our password fields enforcing a 4/5 score or better

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Review by Twixes — APPROVED** [review]
This is great! I love compliance now

File: frontend/src/lib/components/PasswordStrength.tsx [hunk 1]
```diff
@@ -1,26 +1,58 @@
...
+    validatedPassword: ValidatedPasswordResult
 }): JSX.Element {
-    // passwordScore is 0 if no password input
-    // passwordScore is 20, 40, 60, 80, or 100 if password input, based on zxcvbn score (which is 0, 1, 2, 3, or 4)
-    const passwordScore: number = password.length && zxcvbn(password).score * 20 + 20
+    const { score, feedback } = validatedPassword
 
     return (
-        <Tooltip title="Password strength">
+        <Tooltip
```
Comment (zlwaterfield) [inline frontend/src/lib/components/PasswordStrength.tsx:38]: Nit: the tooltip hover area is very small and I'm struggling to get it to show up even knowing it's there. Not a blocker but we may want to try and make the hover surface area larger.

File: frontend/src/lib/components/PasswordStrength.tsx [hunk 2]
```diff
@@ -1,26 +1,58 @@
...
+            title={
+                <>
+                    Your password scores a <strong>{score}/5</strong>
```
Comment (zlwaterfield) [inline frontend/src/lib/components/PasswordStrength.tsx:41]:
```suggestion
                    Your password scores a{" "}
                    <strong className="space-x-0.5">
                        <span>{score}</span>
                        <span>/</span>
                        <span>5</span>
                    </strong>
```

Just a small spacing to make it look more natural

before:
Screenshot 2024-05-09 at 12 44 22 PM

after:
Screenshot 2024-05-09 at 12 44 05 PM

**Review by zlwaterfield — APPROVED** [review]
Thanks for doing this! Tested locally and LGTM.

File: frontend/src/scenes/settings/user/ChangePassword.tsx [hunk 3]
```diff
@@ -26,7 +26,7 @@ export function ChangePassword(): JSX.Element {
                     <div className="flex flex-1 items-center justify-between">
                         <span>Password</span>
                         <span className="w-20">
-                            <PasswordStrength password={changePassword.password} />
+                            <PasswordStrength validatedPassword={validatedPassword} />
```
Comment (xrdt) [inline frontend/src/scenes/settings/user/ChangePassword.tsx:29]: There's no feedback shown on invalid password submission.

Comment (xrdt) [reply frontend/src/scenes/settings/user/ChangePassword.tsx:29]: Sorry, there is. I was looking in the wrong place.

**Review by xrdt — APPROVED** [review]

**Pull request merged**